clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -698,6 +722,16 @@ std::set<ConstraintVariable *> ProgramInfo::getVariable(clang::Decl *D,
   }
 }
 
+std::set<ConstraintVariable *>
+ProgramInfo::getCompoundLiteral(clang::CompoundLiteralExpr *CLE,
+                                clang::ASTContext *C) {
+  VariableMap::iterator I = Variables.find(PersistentSourceLoc::mkPSL(CLE, *C));
```

Comment:
Seems to me that the name `Variables` is not a good one. As you are seeing here, it's really a map from ProgramSourceLocations to ConstraintVariables, which are essentially types; it's the PSLs which might need to get rewritten. When I see `Variables` I am thinking "program variables" (i.e., variables in the original source program).

Ideas for changing the name?

Reply:
I had similar thoughts working on this, but did not come up with anything I felt good enough with to make the change. I'll think on it more as I address your comment.